airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStorageClient.kt

```diff
@@ -38,6 +38,16 @@ interface ObjectStorageClient<T : RemoteObject<*>> {
 }
 
 interface StreamingUpload<T : RemoteObject<*>> {
-    suspend fun uploadPart(part: ByteArray)
+    /**
+     * Uploads a part of the object. Each part must have a unique index. The parts do not need to be
+     * uploaded in order. The index is 1-based.
+     */
+    suspend fun uploadPart(part: ByteArray, index: Int)
+
+    /**
+     * Completes a multipart upload. All parts must be uploaded before completing the upload, and
+     * there cannot be gaps in the indexes. Multiple calls will return the same object, but only the
+     * first call will have side effects.
+     */
```

Comment:
Nit: throw the word "idempotent" in there. Also, why did it need to be idempotent?

Reply:
I was self-conscious about having over-used it lately. I only recently kicked my "orthogonal" habit. :)

Reply:
We're all idempotency junkies over here so it's a safe space.